### ISSUE TITLE
Port number command line argument

### ISSUE THREAD

**Issue by metakermit** [yeoman/generator-webapp_DEPRECATED#524]
Very often, I have multiple projects based on generator-webapp that I want to work on, but I can't start the servers at once, due to a port clash. The _Gruntfile.js_ has a hard-coded port 9000 value.

```
Fatal error: Port 9000 is already in use by another process.
```

It would be really neat if I could simply pass the port number for the dev server to avoid these conflicts

```
grunt serve 9100
```

This would then start the app on [http://localhost:9100](). Other frameworks like Django already have this functionality.

```
./manage.py runserver 9100
```

**Comment by silvenon** [comment]
Thanks for the suggestion, I like it. I'll see when I can implement it.

**Comment by fernoftheandes** [comment]
:+1:

**Comment by sindresorhus** [comment]
Why not let `grunt-contrib-connect` automagically pick the port. Then you don't have to manually do anything at all.

**Comment by silvenon** [comment]
Defaults to 9000, but takes a different one when it's taken?

**Comment by kevva** [comment]
It'll use a random one.

**Comment by sindresorhus** [comment]
Also supports what @silvenon said: https://github.com/gruntjs/grunt-contrib-connect#useavailableport

**Comment by metakermit** [comment]
Also, I don't know what would the values for livereload and the test server be. I'm guessing there's no way to have multiple apps using livereload, so it should probably have to be disabled for the 2nd app? The test port can probably be handled in the same way as the main server - so try 9001 and jump to any open port if it's taken.

**Comment by silvenon** [comment]
@kermit666 I tried running multiple servers and it seems that BrowserSync (we're currently using it) just find another port when it's taken. Can you see if this is the case on your end?

**Comment by silvenon** [comment]
Oh, the BrowserSync version isn't released yet, that explains it. I'll make a release soon.

**Comment by silvenon** [comment]
Released in [v0.6.0](https://github.com/yeoman/generator-webapp/releases/tag/v0.6.0). Closing this as it's an outdated report when we were using connect. Will reopen if the issue is persisting.

**Comment by metakermit** [comment]
@silvenon thanks, just upgraded to the latest package version. Will try it out and reopen if I encounter any problems.
